x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termsenum/action/TransportTermsEnumAction.java

```diff
@@ -413,40 +413,39 @@ private boolean canAccess(
             IndicesAccessControl indicesAccessControl = threadContext.getTransient(AuthorizationServiceField.INDICES_PERMISSIONS_KEY);
             IndicesAccessControl.IndexAccessControl indexAccessControl = indicesAccessControl.getIndexPermissions(shardId.getIndexName());
 
-            if (indexAccessControl != null) {
-                final boolean dls = indexAccessControl.getDocumentPermissions().hasDocumentLevelPermissions();
```

Comment:
If `indexAccessControl` is not `null` AND `indexAccessControl#isGranted` is `false`, then `indexAccessControl.getDocumentPermissions().hasDocumentLevelPermissions()` is also `false` (same for `#hasFieldLevelPermissions()`).
(Though this is only theoretically possible, in practice the code will not reach to this point if `indexAccessControl#isGranted` is `false`).